crates/miden-objects/src/account/storage/header.rs

```diff
@@ -96,25 +102,31 @@ impl AccountStorageHeader {
     ///
     /// # Errors
     /// - If the index is out of bounds.
-    pub fn slot(&self, index: usize) -> Result<&(StorageSlotType, Word), AccountError> {
-        self.slots.get(index).ok_or(AccountError::StorageIndexOutOfBounds {
-            slots_len: self.slots.len() as u8,
-            index: index as u8,
-        })
-    }
+    pub fn slot_header(
```

Comment:
Will this change to take `slot_name` in the future? Currently we construct the name as part of this method's logic rather than passing it in

Reply:
I'm still working this out in the follow-up PR, so tbd :)

Sorry for all the temporary changes :upside_down_face:. This is one of the functions temporarily changed to translate index to name internally, which will definitely go away. Not clear yet whether it will end up taking `SlotName` or `SlotNameId`.